### PR TITLE
Let OOME_ARGS be overridable

### PR DESCRIPTION
**Pull request by nehalrp**

### CODE REVIEW

File: bin/aws-kinesis-agent-manual-run.sh [hunk 1]
```diff
@@ -11,7 +11,7 @@ LIB_DIR="/usr/share/aws-kinesis-agent/lib"
 CLASSPATH="$LIB_DIR":$(find "$LIB_DIR" -type f -name \*.jar | paste -s -d:):"$CLASSPATH"
 
 JAVACMD="java"
-OOME_ARGS="-XX:OnOutOfMemoryError=\"/bin/kill -9 %p\""
+OOME_ARGS="( kill -15 %p && sleep 30 && ps axf | grep aws-kinesis-agent | grep -v grep | awk '{print $1}' | xargs kill -9 & )"
 JVM_ARGS="-server -Xms${JAVA_START_HEAP} -Xmx${JAVA_MAX_HEAP} $JVM_ARGS"
```
Comment (diranged) [inline bin/aws-kinesis-agent-manual-run.sh]: I recommend pgep here

Comment (nehalrp) [inline bin/aws-kinesis-agent-manual-run.sh]: It is ideal, but would not work in our case:
`[root@staging-us1-ecs-uswest2-i-066245814:~]# pgrep aws-kinesis-agent
[root@staging-us1-ecs-uswest2-i-066245814:~:1]# pgrep java
46792
97659
119421`

Comment (nehalrp) [inline bin/aws-kinesis-agent-manual-run.sh]: It is ideal, but would not work in our case:
```
[root@staging-us1-ecs-uswest2-i-066245814:~]# pgrep aws-kinesis-agent
[root@staging-us1-ecs-uswest2-i-066245814:~:1]# pgrep java
46792
97659
119421
```

Comment (stlava) [inline bin/aws-kinesis-agent-manual-run.sh]: per our discussion `grep -v grep` would match the `grep -v grep` in the command

Comment (nehalrp) [inline bin/aws-kinesis-agent-manual-run.sh]: ah seems like `pgrep` has a full argument command. this should address it

**Comment by nehalrp** [comment]
Seems like we should have a check to make sure we aren't killing the respawned proc.

File: bin/aws-kinesis-agent-manual-run.sh (hunk 1 above)
Comment (diranged) [inline bin/aws-kinesis-agent-manual-run.sh]: Use pgrep -f

**Comment by stlava** [comment]
This isn't quite right. The evaluation of `[ \"%p\" == \"$(pgrep -f 'aws-kinesis-agent')\" ]]` is done before the process starts and it renders to `[[ "%p" == "" ]]`. For example:

```
#
# java.lang.OutOfMemoryError: Java heap space
# -XX:OnOutOfMemoryError="( kill -15 %p && sleep 30 && [[ "%p" == "" ]] && pgrep -f 'aws-kinesis-agent' | xargs kill -9 & )"
#   Executing "( kill -15 11733 && sleep 30 && [[ "11733" == "" ]] && pgrep -f 'aws-kinesis-agent' | xargs kill -9 & )"...
```

```
  502 11733 10554   0  4:54PM ttys001    0:14.99 /usr/bin/java -server -Xms32m -Xmx64m -Dcom.sun.management.jmxremote -XX:OnOutOfMemoryError=( kill -15 %p && sleep 30 && [[ "%p" == "" ]] && pgrep -f 'aws-kinesis-agent' | xargs kill -9 & ) -Dcom.sun.management.jmxremote.port=4143 -Dcom.sun.management.jmxremote.rmi.port=4143 -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -Dlog4j.configuration=file://log4j.xml -cp amazon-kinesis-agent/dependencies/*:/usr/share/java/*:amazon-kinesis-agent/ant_build/lib/AWSKinesisStreamingDataAgent-1.1.jar: com.amazon.kinesis.streaming.agent.Agent
```

**Pull request merged**